### PR TITLE
Add default override indicator

### LINKED ISSUE #137943: Show the extension/source overriding the default value of a setting in settings editor
With https://github.com/microsoft/vscode/issues/46851, an extension can override default values of a setting by contributing defaults to `contributeDefaults` contribution point. Eg:

```json
"configurationDefaults": {
      "workbench.colorTheme": "Monokai"
    }
```

It would be nice if settings editor can display the source of the default value

image

CC: @misolori

### PR DESCRIPTION
**Pull request by rzhao271**
Fixes #137943

Demo

And an example of how it looks with both the default overridden and the sync ignored:
override-ignore-sync

### CODE REVIEW

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts [hunk 1]
```diff
@@ -943,12 +954,22 @@ export abstract class AbstractSettingRenderer extends Disposable implements ITre
...
+				} else if (typeof defaultValueSource === 'string') {
+					template.defaultOverrideIndicator.title = localize('defaultOverridenDetails', "Default overriden by {0}", defaultValueSource);
+				}
+			}
 		};
-		update();
+		updateTitleElements();
 		template.elementDisposables.add(this.onDidChangeIgnoredSettings(() => {
-			update();
+			updateTitleElements();
```
Comment (rzhao271) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:972]: TODO: only the sync ignore element needs to be updated in this case.

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts [hunk 2]
```diff
@@ -781,6 +782,14 @@ export abstract class AbstractSettingRenderer extends Disposable implements ITre
 		return syncIgnoredElement;
 	}
 
+	protected createDefaultOverrideIndicator(container: HTMLElement): HTMLElement {
+		const defaultOverrideIndicator = DOM.append(container, $('span.setting-item-default-overriden'));
+		const defaultOverrideLabel = new SimpleIconLabel(defaultOverrideIndicator);
+		defaultOverrideLabel.text = `($(alert) ${localize('defaultOverridenLabel', "Default overriden")})`;
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:788]:
```suggestion
		defaultOverrideLabel.text = `($(alert) ${localize('defaultOverridenLabel', "Default overridden")})`;
```

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts [hunk 3]
```diff
@@ -943,12 +954,22 @@ export abstract class AbstractSettingRenderer extends Disposable implements ITre
...
 
-		const update = () => {
+		const defaultValueSource = element.setting.defaultValueSource;
+		const updateTitleElements = () => {
 			template.syncIgnoredElement.style.display = this.ignoredSettings.includes(element.setting.key) ? 'inline' : 'none';
+			template.defaultOverrideIndicator.style.display = 'none';
+			if (defaultValueSource) {
+				template.defaultOverrideIndicator.style.display = 'inline';
+				if (typeof defaultValueSource !== 'string' && defaultValueSource.id !== element.setting.extensionInfo?.id) {
+					template.defaultOverrideIndicator.title = localize('defaultOverridenDetails', "Default overriden by {0}", defaultValueSource.displayName ?? defaultValueSource.id);
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:964]:
```suggestion
					template.defaultOverrideIndicator.title = localize('defaultOverridenDetails', "Default overridden by {0}", defaultValueSource.displayName ?? defaultValueSource.id);
```

File: src/vs/workbench/contrib/preferences/browser/settingsTree.ts [hunk 4]
```diff
@@ -943,12 +954,22 @@ export abstract class AbstractSettingRenderer extends Disposable implements ITre
...
+				} else if (typeof defaultValueSource === 'string') {
+					template.defaultOverrideIndicator.title = localize('defaultOverridenDetails', "Default overriden by {0}", defaultValueSource);
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/preferences/browser/settingsTree.ts:966]:
```suggestion
					template.defaultOverrideIndicator.title = localize('defaultOverridenDetails', "Default overridden by {0}", defaultValueSource);
```

**Review by sandy081 — APPROVED** [review]
LGTM

**Review by raghavthind2005 — APPROVED** [review]

**Review by sandy081 — APPROVED** [review]

**Pull request merged**